main/src/library/Fixpoint/Ram/RelOp.flix

```diff
@@ -0,0 +1,48 @@
+/*
+ * Copyright 2021 Benjamin Dahse
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+namespace Fixpoint/Ram {
+    pub enum RelOp[v] {
+        case Search(RowVar, RelSym[v], RelOp[v])
+        case Query(RowVar, RelSym[v], Array[RamTerm[v]], RelOp[v])
+        case Project(Array[RamTerm[v]], RelSym[v])
+        case If(List[BoolExp[v]], RelOp[v])
```

Comment:
Is there any reason to favor `List` over `Array`? The latter probably has better locality.

Reply:
I rarely know the length of the list upfront and conditionally need to cons onto it.